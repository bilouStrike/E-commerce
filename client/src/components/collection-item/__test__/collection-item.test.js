import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';

import { CollectionItem } from '../collection-item.component';

describe('Collection Item', () =>{
    const MockItem = { 
        name: 'item',
        price: 50,
        imageUrl: 'image url'  
    };
    const mockAddtoCart = jest.fn();
    let wrapper = shallow(<CollectionItem item={MockItem} addItemToCart={mockAddtoCart}/>);

    it('Expect to invoke additemToCart on click in CustomButton', () => {
        wrapper.find('CustomButton').simulate('click');
        expect(mockAddtoCart).toHaveBeenCalled();
    });

    it('Expect to render CollectionItem component', () => {
        wrapper = toJson(wrapper);
        expect(wrapper).toMatchSnapshot();
    });

});