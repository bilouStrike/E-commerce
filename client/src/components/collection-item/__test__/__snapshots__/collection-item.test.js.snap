// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection Item Expect to render CollectionItem component 1`] = `
<div
  className="collection-item"
>
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url(image url)",
      }
    }
  />
  <div
    className="collection-footer"
  >
    <span
      className="name"
    >
      item
    </span>
    <span
      className="price"
    >
      50
    </span>
  </div>
  <CustomButton
    inverted={true}
    onClick={[Function]}
  >
     
    Add to cart
     
  </CustomButton>
</div>
`;
